sdk/webpubsub/Azure.Messaging.WebPubSub/src/HealthApiClient_extensions.cs

```diff
@@ -0,0 +1,42 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+// <auto-generated/>
+
+#nullable disable
+
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+using Azure;
+using Azure.Core;
+using Azure.Core.Pipeline;
+
+#pragma warning disable AZC0007
+
+namespace Azure.Messaging.WebPubSub
+{
+    /// <summary> The HealthApi service client. </summary>
+    internal partial class HealthApiClient
+    {
+        /// <summary> Initializes a new instance of HealthApiClient. </summary>
+        /// <param name="credential"> A credential used to authenticate to an Azure Service. </param>
+        /// <param name="endpoint"> server parameter. </param>
+        /// <param name="options"> The options for configuring the client. </param>
+        public HealthApiClient(AzureKeyCredential credential, Uri endpoint = null, WebPubSubServiceClientOptions options = null)
+        {
+            if (credential == null)
+            {
+                throw new ArgumentNullException(nameof(credential));
+            }
+            endpoint ??= new Uri("");
+
+            options ??= new WebPubSubServiceClientOptions();
+            _clientDiagnostics = new ClientDiagnostics(options);
+            var authPolicy = new WebPubSubAuthenticationPolicy(credential);
+            Pipeline = HttpPipelineBuilder.Build(options, new HttpPipelinePolicy[] { authPolicy, new LowLevelCallbackPolicy() });
```

Comment:
What kind of auth are we doing here?  If it's any sort of message signing, we should probably do it after the callback.